app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt

```diff
@@ -158,174 +185,33 @@ fun AccountDetails(
-            socials.forEachIndexed { index, userSocial ->
-              InputChip(
-                  label = { Text(userSocial.social.name) },
-                  onClick = {},
-                  selected = true,
-                  modifier =
-                      Modifier.padding(3.dp)
-                          .testTag(AccountDetailsTestTags.SOCIALS_CHIP + userSocial.social.title),
-                  avatar = {
-                    Icon(
-                        Icons.Default.Close,
-                        contentDescription =
-                            context.getString(R.string.account_details_content_description_close),
-                        modifier =
-                            Modifier.clickable {
-                              userSocialsFlow.value =
-                                  userSocialsFlow.value.toMutableList().apply { removeAt(index) }
-                            })
-                  })
-            }
-          }
+          TextFields(
+              isErrors,
+              firstName,
+              lastName,
+              bio,
+              { firstName = it },
+              { lastName = it },
+              { bio = it })
+
+          ProfilePicturePicker(profilePictureUri, { profilePictureUri.value = Uri.EMPTY })
+
+          OverlayButtonsAndFlowRows(
+              interestsFlow,
+              userSocialsFlow,
+              { showInterestsOverlay = true },
+              { showSocialsOverlay = true })
+
           Button(
               modifier = Modifier.testTag(AccountDetailsTestTags.CONTINUE_BUTTON),
-              onClick = {
-                if (profilePictureUri.value == Uri.EMPTY) {
-                  createUser("")
-                } else {
-                  val inputStream = context.contentResolver.openInputStream(profilePictureUri.value)
-
-                  imageRepository.uploadImage(
-                      inputStream!!,
-                      "images/users/${userId}",
-                      onSuccess = createUser,
-                      onFailure = { exception ->
-                        Log.e("AccountDetails", "Error uploading image: $exception")
-                        Toast.makeText(
-                                context,
-                                context.getString(R.string.account_details_image_upload_error),
-                                Toast.LENGTH_SHORT)
-                            .show()
-                      })
-                }
-              }) {
+              /**
+               * The [onCreateUser] function is called with the [profilePictureUri] and [createUser]
+               * lambda function as parameters. [createUser] calls upon another lambda that uploads
+               * the user if all the required fields are filled. This method hierarchy is necessary
+               * due to the fact that uploading an image needs to know if the URI is empty or not
+               * before creating the User.
+               */
```

Comment:
I don't think it's good practice to put javadoc in the body of a function, you could make it a simple comment or you could create a javadoc for AccountDetailsContent and put this explanation in it. But it's a good initiative to explain in detail this pretty complex process :)

Reply:
You're right but because it is a lambda and not a "function" so to say, I didn't really know where to put it